nomad/deploymentwatcher/deployment_watcher.go

```diff
@@ -137,7 +166,7 @@ func (w *deploymentWatcher) SetAllocHealth(
 			}
 
 			// Check if the group has autorevert set
-			group, ok := w.d.TaskGroups[alloc.TaskGroup]
+			group, ok := w.getDeployment().TaskGroups[alloc.TaskGroup]
```

Comment:
Will TaskGroups always be initialized on the deployment that is returned from `getDeployment`?

Reply:
Yes